source/common/http/filter_manager.cc

```diff
@@ -183,6 +183,11 @@ bool ActiveStreamFilterBase::commonHandleAfterDataCallback(FilterDataStatus stat
       buffer_was_streaming = status == FilterDataStatus::StopIterationAndWatermark;
       commonHandleBufferData(provided_data);
     } else if (complete() && !hasTrailers() && !bufferedData()) {
+      if (parent_.state_.destroyed_) {
```

Comment:
nit: can you check put `&& !parent_.state_.destroyed_` above and avoid this extra if?

Reply:
Done. Removed comment.